src/featureManager.ts

```diff
@@ -75,14 +73,15 @@ export class FeatureManager {
         return !shortCircuitEvaluationResult;
     }
 
+    async #getFeatureFlag(featureName: string): Promise<any> {
+        const featureFlag = await this.#provider.getFeatureFlag(featureName);
+        validateFeatureFlag(featureFlag);

```

Comment:
It makes more sense to let feature provider validate feature flag. Let's move this to feature provider.

Reply:
Validating by manager can also handle custom filter scenarios.

Reply:
Did you add any testcases for that?

Reply:
Ok. let's not validate the specific configuration of a feature filter and move the validation to feature flag provider.